axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineComputeAnalyticServiceImpl.java

```diff
@@ -183,6 +175,28 @@ public MoveLine selectDefaultDistributionTemplate(MoveLine moveLine) throws Axel
     return moveLine;
   }
 
+  protected AnalyticDistributionTemplate getDistributionTemplate(
+      MoveLine moveLine, Account account, TradingName tradingName) throws AxelorException {
+    AnalyticDistributionTemplate analyticDistributionTemplate = null;
+    if (account != null && account.getAnalyticDistributionAuthorized()) {
+      if (account.getAnalyticDistributionTemplate() != null
+          && accountConfigService
+                  .getAccountConfig(account.getCompany())
+                  .getAnalyticDistributionTypeSelect()
+              == AccountConfigRepository.DISTRIBUTION_TYPE_PRODUCT) {
+        analyticDistributionTemplate = moveLine.getAccount().getAnalyticDistributionTemplate();
+      } else if (tradingName != null
+          && tradingName.getAnalyticDistributionTemplate() != null
+          && accountConfigService
+                  .getAccountConfig(account.getCompany())
+                  .getAnalyticDistributionTypeSelect()
+              == AccountConfigRepository.DISTRIBUTION_TYPE_TRADING_NAME) {
+        analyticDistributionTemplate = tradingName.getAnalyticDistributionTemplate();
+      }
+    }
```

Comment:
please use guard clause instead of nesting `if`
I think the `moveLine` is unnecessary in this method (`moveLine.getAccount()` must be replaced by `account`)